sklearn/utils/estimator_checks.py

```diff
@@ -183,6 +184,10 @@ def _yield_classifier_checks(classifier):
     # test if predict_proba is a monotonic transformation of decision_function
     yield check_decision_proba_consistency
 
+    if isinstance(classifier, LinearClassifierMixin):
+        if "class_weight" in classifier.get_params().keys():
+            yield check_class_weight_balanced_linear_classifier
```

Comment:
@OmarManzoor 
yes, the test was present in this file, but not a part of `check_estimator` function or `parametrize_with_checks`. So for third party estimators this was not enabled, and our estimators also wouldn't be able to skip it with `_xfail_checks`. This moves it here like all the other tests.